deploy/tasks/pre-build.yaml

```diff
@@ -0,0 +1,145 @@
+metadata:
+  name: pre-build
+  annotations:
+    tekton.dev/pipelines.minVersion: 0.12.1
+    tekton.dev/tags: image-build, konflux
+  labels:
+    app.kubernetes.io/version: "0.1"
+    build.appstudio.redhat.com/build_type: docker
+spec:
+  description: |-
+    Sets up pre-build running the preprocessor, pushing the source and creating the OCI image.
+  params:
+    - name: IMAGE_URL
+      description: URL of the OCI image to use.
+      type: string
+    - name: NAME
+      description: Name of the pipeline run (i.e. unique dependency build name)
+      type: string
+    - name: GIT_SCRIPT
+      description: Git clone commands
+      type: string
+    - name: GIT_IDENTITY
+      description: Git username
+      type: string
+    - name: GIT_URL
+      description: URL to determine whether we're using gitlab or github
+      type: string
+    - name: GIT_DEPLOY_TOKEN
+      description: Name of jvm-build-git-repo-secrets secret containing git password/token to use.
+      type: string
+    - name: GIT_SSL_VERIFICATION
+      description: Whether to disable ssl verification
+      type: string
+      default: "false"
+    - name: GIT_REUSE_REPOSITORY
+      description: Whether to reuse existing git repository or create new one
+      type: string
+    - name: SCM_URL
+      description: Reference to the git repository
+      type: string
+    - name: SCM_HASH
+      description: Git hash
+      type: string
+    - name: RECIPE_IMAGE
+      description: The image from the build recipe to use
+    - name: BUILD_SCRIPT
+      description: The build script to embed with the Containerfile
+    - name: PREPROCESSOR_ARGS
+      description: The arguments for the build preprocessor
+    - name: ORAS_OPTIONS
+      type: string
+      description: Optional environment variable string for build-trusted-artifacts
+      default: ""
+    - name: JVM_BUILD_SERVICE_REQPROCESSOR_IMAGE
+      description: Name of the processor image. Useful to override for development.
+      type: string
+      default: "quay.io/redhat-appstudio/hacbs-jvm-build-request-processor:dev"
+  results:
+    - name: PRE_BUILD_IMAGE_DIGEST
+      description: Digest of the image just built
+    - name: GIT_ARCHIVE
+      description: Git archive information
+  workspaces:
+    - description: The git repo will be cloned onto the volume backing this Workspace.
+      name: source
+      mountPath: /var/workdir
+    - name: tls
+  steps:
+    # Should we use our own git clone task? Or embed (somehow) Konflux's version?
+    - name: git-clone
+      image: $(params.RECIPE_IMAGE)
+      computeResources:
+        limits:
+          cpu: 300m
+          memory: 512Mi
+        requests:
+          cpu: 10m
+          memory: 512Mi
+      securityContext:
+        runAsUser: 0
+      env:
+        - name: GIT_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: jvm-build-git-secrets
+              key: .git-credentials
+      script: |
+        $(params.GIT_SCRIPT)
```

Comment:
Currently **not** using Konflux git clone task but the original simple git clone from JBS and passing it through as a script from the operator.

Reply:
The more we use Konflux's tasks the better I think, but not a problem for now to use our own. Can be a subsequent PR.